### DISCUSSION TITLE
How should we respond back to mercadopago webhook?

### DISCUSSION THREAD

**Discussion by ihojmanb** [mercadopago/sdk-js#114] [Q&A] [answered]
I'm setting a `notification_url` in my payment function to be able to by notified whenever a payment event happens, through mercadopago's webhooks, as noted [here](https://www.mercadopago.com.mx/developers/es/docs/subscriptions/additional-content/notifications/webhooks#bookmark_acciones_necesarias_despu%C3%A9s_de_recibir_la_notificaci%C3%B3n).

I understand that mercadopago will hit de `notification_url` endpoint and will wait until I respond back to the server with an `HTTP STATUS 200`. If no response is heard, mercadopago will hit the `notification_url` over and over again.

My question is, how do I send to mercadopago my confirmation response back? I would expect to receive an endpoint where I could tell mercadopago 'hey, I received your message' with a POST request, but there is nothing like it in the mercadopago message.

**Comment by ihojmanb** [comment, marked as answer]
To be more specific, I'm receiving two kinds of notifications:

1. the one with the payment information, with the following body:
```
 {
  action: 'payment.created',
  api_version: 'v1',
  data: { id: '52090698612' },
  date_created: '2022-12-01T23:14:47Z',
  id: 104321540926,
  live_mode: true,
  type: 'payment',
  user_id: '1188717534'
}
```
this one has the following `User-Agent`: `'MercadoPago WebHook v1.0 payment'`

2. another with the resource information, with the following body:
```
{
  resource: 'https://api.mercadolibre.com/collections/notifications/52099698668',
  topic: 'payment'
}
```
this one has the following `User-Agent`: `'MercadoPago Feed v2.0 payment'`

number `1` gives me all the data that I need. number `2` I don't get what is trying to say to me.

That said, because I don't know how to tell mercadopago that I've got the notification, it keeps sending me both messages. How can I stop this behaviour?

**Reply by gdeandradero** [reply]
Hello, your confirmation response is your status code 200 or 201.
Example: mercadopago hits your notification_url, if your notification_url answer 200 or 201, mercadopago will know that you receives the message successfully.

You are receiving two messages because we send two messages: the first is the Webhook and the second is the IPN.
If you want to receive notifications only from Webhook and not from IPN, you can add in the notification_url the parameter source_news=webhook. For example: https://www.yourserver.com/notifications?source_news=webhooks

**Reply by ihojmanb** [reply]
I'm such an idiot! I wasn't realizing that _my_ endpoint was the one to return a response to the mercadopago webhook request. So embarrassing jajaja. Thanks @gdeandradero for clarifying things 😄

**Reply by gdeandradero** [reply]
hahaha no problem, sometimes some things go unnoticed
if you need anything else just say 😄

**Reply by ihojmanb** [reply]
💕 💕 💕
